packages/features/ee/organizations/components/AddNewTeamsForm.tsx

```diff
@@ -73,41 +126,76 @@ export const AddNewTeamsForm = () => {
         showToast(t("duplicated_slugs_warning", { slugs: data.duplicatedSlugs.join(", ") }), "warning");
         // Server will return array of duplicated slugs, so we need to wait for user to read the warning
         // before pushing to next page
-        setTimeout(() => {
-          router.push(`/event-types`);
-        }, 3000);
+        setTimeout(() => handleSuccessRedirect, 3000);
       } else {
-        router.push(`/event-types`);
+        handleSuccessRedirect();
+      }
+
+      function handleSuccessRedirect() {
+        if (allowWizardCompletionWithoutUpgrading) {
+          router.push(`/event-types`);
+          return;
+        }
+        publishOrgMutation.mutate();
       }
     },
     onError: (error) => {
       showToast(t(error.message), "error");
     },
   });
 
-  const handleInputChange = (index: number, event: any) => {
-    const { name, value } = event.target;
-    setValue(`teams.${index}.name`, value);
-    trigger(`teams.${index}.name`);
-  };
-
   const handleFormSubmit = () => {
-    if (formState.isValid) {
```

Comment:
`isValid` doesn't work correctly sometimes blocking the form submission even when the submit button is enabled. So, removed it.